decide/src/main/java/group17/LicAnalyzer.java

```diff
@@ -276,7 +276,44 @@ public boolean lic11(InputHandler input) {
     }
 
     public boolean lic12(InputHandler input) {
-        return false;
+
+        if (input.NUMPOINTS < 2 || input.NUMPOINTS > 100) {
+            throw new IllegalArgumentException("NUMPOINTS must be inside the range [2, 100]");
+        }
+
+        if (input.NUMPOINTS == 2) {
+            return false;
+        }
+
+        if (input.K_PTS < 1 || input.K_PTS > input.NUMPOINTS-2) {
+            throw new IllegalArgumentException("K_PTS must be between 1 (inclusive) and NUMPOINTS-2 (inclusive)");
+        }
+
+        if (input.LENGTH1 < 0) {
+            throw new IllegalArgumentException("LENGTH1 must be a positive number");
+        }
+
+        if (input.LENGTH2 < 0) {
+            throw new IllegalArgumentException("LENGTH2 must be a positive number");
+        }
+
+        boolean distGreaterThanLENGTH1 = false;
+        boolean distLessThanLENGTH2 = false;
+
+        for (int i = 0; i < input.NUMPOINTS - input.K_PTS - 1; i++) {
```

Comment:
Possibility for arrayIndexOutOfBounds here, depending on K_PTS. For example, K_PTS == NUMPOINTS would cause an exception.